### PR TITLE
Implement token request and first store in secret

### PR DESCRIPTION
**Pull request by bastjan**
Too much code - as always with the initial implementation. Sorry.

* Adds a CRD+Controller to create emergency account tokens
* The controller:
  * Reconciles a SA with the same name as the CRD
  * Requests tokens if not enough verified tokens exists
  * Stores those tokens
  * Verifies those tokens, using TokenReview
 
The storage implementation is generic through `StoreToken`, if the implementation supports `RetrieveToken` the token is periodically checked against Kubernetes.

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.

### CODE REVIEW

File: api/v1beta1/emergencyaccount_types.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+type SecretStoreSpec struct {
+}
+
+// LogStoreSpec outputs the token to the log but does not store it anywhere
+type LogStoreSpec struct {
+}
+
+// TokenStatus defines the observed state of the managed token
+type TokenStatus struct {
+	// UUID is the unique identifier of the token.
```
Comment (simu) [inline api/v1beta1/emergencyaccount_types.go:77]:
```suggestion
	// UID is the unique identifier of the token.
```

File: api/v1beta1/emergencyaccount_types.go [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+	// Currently only user for error messages.
+	UID types.UID `json:"uid,omitempty"`
+	// Ref is a reference to the token. The used storage should be able to uniquely identify the token
```
Comment (simu) [inline api/v1beta1/emergencyaccount_types.go:80]: Looks like this comment may be outdated. The field is called `Refs`

**Review by simu — APPROVED** [review]
LGTM overall. Some inline comments for probaby outdated comments.

File: api/v1beta1/emergencyaccount_types.go (hunk 1 above)
Comment (bastjan) [reply api/v1beta1/emergencyaccount_types.go]: Fixed and improved comments in general.

**Pull request merged**